samples/opengl/depthMapTriangulation/decimate.cpp

```diff
@@ -0,0 +1,694 @@
+/**
+ * Copyright (c) 2020 Simon Mederer
+ * Licensed under the MIT License.
+ * See LICENSE file for more information.
+ */
+
+#include "decimate.h"
+
+// --- PUBLIC ---
+
+Saiga::QuadricDecimater::QuadricDecimater(Settings const& s) : settings(s) {}
+
+Saiga::QuadricDecimater::~QuadricDecimater() {}
+
+void Saiga::QuadricDecimater::decimate_quadric(MyMesh& mesh_in)
+{
+    mesh = mesh_in;
```

Comment:
Is this a deep copy? If yes explain why

Reply:
My intention is to copy the mesh into my decimater object so that all methods within it can simply access that and don't need to get the mesh as a parameter. Is that reasonable or should I change it?

Reply:
I changed the `QuadricDecimater::mesh` to a pointer since it only gets used by `QuadricDecimater::decimate_quadric` and the methods called from within there. This way I don't copy the mesh, but it might also be bad code